subprojects/core/src/integTest/groovy/org/gradle/api/file/SettingsLayoutIntegrationTest.groovy

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.file
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import org.gradle.integtests.fixtures.build.BuildTestFile
+
+class SettingsLayoutIntegrationTest extends AbstractIntegrationSpec {
+    private String printLocations() {
+        groovyScript """
+            println "settings root dir: " + layout.rootDirectory + "."
+            println "settings dir: " + layout.settingsDirectory + "."
+            println "settings source file: " + layout.dir(providers.provider { buildscript.sourceFile }).get() + "."
+            println "settings relative location: " + layout.file(providers.provider { new File("somefile.txt") }).get() + "."
+        """
+    }
+
+    def "layout is available for injection"() {
+        settingsFile """
+            import javax.inject.Inject
+            import org.gradle.api.file.FileSystemLayout
+            import org.gradle.api.provider.ProviderFactory
+            import org.gradle.api.initialization.Settings
```

Comment:
💭 TIL, none of these imports are necessary for the test to work, and the imports could be simply removed.

Reply:
Indeed... I was wondering if it could be needed to help with syntax-highlighting/navigation from the IDE, ProjectLayoutIntegrationTest does not have them and navigation works fine.